packages/grafana-data/src/field/fieldOverrides.ts

```diff
@@ -358,20 +360,37 @@ export function validateFieldConfig(config: FieldConfig) {
   }
 }
 
+const defaultInternalDataLinkSupplier: InternalDataLinkSupplier = (options) => {
+  // For internal links at the moment only destination is Explore.
+  const { link, dataLinkScopedVars, field, replaceVariables } = options;
+  if (!link.internal) {
+    return undefined;
+  }
+  return mapInternalLinkToExplore({
+    link,
+    internalLink: link.internal,
+    scopedVars: dataLinkScopedVars,
+    field,
+    range: link.internal.range ?? ({} as any),
+    replaceVariables,
+  });
+};
```

Comment:
Yes, it supports clicking internal data links in Dashboards (internal data links are attached to the data frame by data source plugins). I just moved the existing logic and wrapped it so it works like a supplier/post-processor.

I agree that ideally, we should not have anything Explore-specific here. If Explore was an application, I can imagine there was a "data link extension point" that would allow passing a post processor to the link. Maybe something worth exploring as a follow up?